tml/org.testeditor.tml.dsl/src/org/testeditor/tml/dsl/Tml.xtext

```diff
@@ -1,10 +1,152 @@
-grammar org.testeditor.tml.dsl.Tml with org.testeditor.tcl.dsl.Tcl 
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+grammar org.testeditor.tml.dsl.Tml with org.eclipse.xtext.xbase.Xbase 
 
 import "http://www.testeditor.org/tml"
+import "http://www.testeditor.org/tsl" as tsl
 import "http://www.testeditor.org/aml" as aml
 import "http://www.eclipse.org/emf/2002/Ecore" as ecore
 
 TmlModel:
 	'package' package=QualifiedName
 	(importSection=XImportSection)?
+	'#' name=ID
+	(macros+=Macro)*
 ;
+
+Macro:

```

Comment:
It would be nice to have something that wraps the macro, such as:
```
## myMacro 
   template = "start with" ${startparam}
    Component: MyComponent
      - put @startparam into <other>
```